docs/interactivity.md

```diff
@@ -0,0 +1,483 @@
+                "description": "Send a customized emotional response survey to myself",
+                "call": {
+                    "url": "http://localhost:8065/plugins/com.mattermost.apps/hello/send"
+                }
+            },
+            {
+                "app_id": "hello",
+                "location_id": "send",
+                "label": "Survey a user",
+                "hint": "Send survey to a user",
+                "description": "Send a customized emotional response survey to a user",
+                "call": {
+                    "url": "http://localhost:8065/plugins/com.mattermost.apps/hello/send",
+                    "type": "form",
+                    "expand": {
+                        "app": "",
+                        "acting_user": "",
+                        "post": "All"
+                    }
+                }
+            }
+        ]
+    },
+    {
+        "location_id": "/command",
+        "bindings": [
+            {
+                "app_id": "hello",
+                "label": "message",
+                "hint": "[--user] message",
+                "description": "send a message to a user",
+                "call": {
+                    "url": "http://localhost:8065/plugins/com.mattermost.apps/hello/send"
+                }
+            },
+            {
+                "app_id": "hello",
+                "location_id": "manage",
+                "hint": "subscribe | unsubscribe ",
+                "description": "manage channel subscriptions to greet new users",
+                "bindings": [
+                    {
+                        "app_id": "hello",
+                        "label": "subscribe",
+                        "hint": "[--channel]",
+                        "description": "subscribes a channel to greet new users",
+                        "call": {
+                            "url": "http://localhost:8065/plugins/com.mattermost.apps/hello/subscribe",
+                            "values": {
+                                "mode": "on"
+                            }
+                        }
+                    },
+                    {
+                        "app_id": "hello",
+                        "label": "unsubscribe",
+                        "hint": "[--channel]",
+                        "description": "unsubscribes a channel from greeting new users",
+                        "call": {
+                            "url": "http://localhost:8065/plugins/com.mattermost.apps/hello/subscribe",
+                            "values": {
+                                "mode": "off"
+                            }
+                        }
+                    }
+                ]
+            }
+        ]
+    }
+]
+```
+
+</details>
+
+<details><summary>Request to App server</summary>
+
+`GET` http://localhost:8065/plugins/com.mattermost.apps/hello/bindings?acting_user_id=f88nesmr7ifhicr8wwf94oxiwa&channel_id=b77x9mu8xindzxkgem8guadsra
```

Comment:
Currently the browser is providing URL params `user_id=f88nesmr7ifhicr8wwf94oxiwa&channel_id=b77x9mu8xindzxkgem8guadsra&scope=webapp`. The `user_id` is obviously unnecessary.

The App server is receiving params `acting_user_id=f88nesmr7ifhicr8wwf94oxiwa&channel_id=b77x9mu8xindzxkgem8guadsra`. Should we be sending more information? Possibly `team_id`, which could come straight from the browser, though there should be some vetting of these values on the proxy plugin side before sending to the App server.

The proxy plugin is expecting these values from the browser:

https://github.com/mattermost/mattermost-plugin-apps/blob/d2526bdbaef440095fbf5d7d0ee355a882b04e0f/server/http/restapi/bindings.go#L12-L18